src/librustdoc/clean/mod.rs

```diff
@@ -1980,6 +1980,15 @@ impl<'tcx> ContainerTy<'tcx> {
 
                 let generics = tcx.generics_of(container);
                 let param = generics.params[index].def_id;
+
+                match tcx.def_kind(param) {
+                    DefKind::TyParam => {}
+                    // The `DefKind` of self type parameter can be `Trait` or `TraitAlias` since it reuses the
+                    // `DefId` of the trait (alias). It doesn't have a meaningful object lifetime default.
+                    DefKind::Trait | DefKind::TraitAlias => return ObjectLifetimeDefault::Empty,
```

Comment:
#101214 has to do the same before calling `object_lifetime_default`. That's where I took the inspiration